docs/sources/flow/reference/components/discovery.ovhcloud.md

```diff
@@ -0,0 +1,165 @@
+* `__meta_ovhcloud_vps_monitoring_ip_blocks`: the monitoring IP blocks of the server.
+* `__meta_ovhcloud_vps_name`: the name of the server.
+* `__meta_ovhcloud_vps_netboot_mode`: the netboot mode of the server.
+* `__meta_ovhcloud_vps_offer_type`: the offer type of the server.
+* `__meta_ovhcloud_vps_offer`: the offer of the server.
+* `__meta_ovhcloud_vps_state`: the state of the server.
+* `__meta_ovhcloud_vps_vcore`: the number of virtual cores of the server.
+* `__meta_ovhcloud_vps_version`: the version of the server.
+* `__meta_ovhcloud_vps_zone`: the zone of the server.
+
+[Dedicated servers][] meta labels:
+* `__meta_ovhcloud_dedicated_server_commercial_range`: the commercial range of the server.
+* `__meta_ovhcloud_dedicated_server_datacenter`: the datacenter of the server.
+* `__meta_ovhcloud_dedicated_server_ipv4`: the IPv4 of the server.
+* `__meta_ovhcloud_dedicated_server_ipv6`: the IPv6 of the server.
+* `__meta_ovhcloud_dedicated_server_link_speed`: the link speed of the server.
+* `__meta_ovhcloud_dedicated_server_name`: the name of the server.
+* `__meta_ovhcloud_dedicated_server_os`: the operating system of the server.
+* `__meta_ovhcloud_dedicated_server_rack`: the rack of the server.
+* `__meta_ovhcloud_dedicated_server_reverse`: the reverse DNS name of the server.
+* `__meta_ovhcloud_dedicated_server_server_id`: the ID of the server.
+* `__meta_ovhcloud_dedicated_server_state`: the state of the server.
+* `__meta_ovhcloud_dedicated_server_support_level`: the support level of the server.
+
+## Component health
+
+`discovery.ovhcloud` is only reported as unhealthy when given an invalid
+configuration. In those cases, exported fields retain their last healthy
+values.
+
+## Debug information
+
+`discovery.ovhcloud` does not expose any component-specific debug information.
+
+## Debug metrics
+
+`discovery.ovhcloud` does not expose any component-specific debug metrics.
+
+## Example
+
+```river
+discovery.ovhcloud "example" {
+	application_key    = APPLICATION_KEY
+	application_secret = APPLICATION_SECRET
+	consumer_key       = CONSUMER_KEY
+	service            = SERVICE
+}
+
+prometheus.scrape "demo" {
+	targets    = discovery.ovhcloud.example.targets
+	forward_to = [prometheus.remote_write.demo.receiver]
+}
+
+prometheus.remote_write "demo" {
+	endpoint {
+		url = PROMETHEUS_REMOTE_WRITE_URL
+		basic_auth {
+			username = USERNAME
+			password = PASSWORD
+		}
+	}
+}
+```
+
+Replace the following:
+  - `APPLICATION_KEY`: The OVHcloud [API][] application key.
+  - `APPLICATION_SECRET`: The OVHcloud [API][] application secret.
+  - `CONSUMER_KEY`: The OVHcloud [API][] consumer key.
+  - `SERVICE`: The OVHcloud service of the targets to retrieve.
+  - `PROMETHEUS_REMOTE_WRITE_URL`: The URL of the Prometheus remote_write-compatible server to send metrics to.
+  - `USERNAME`: The username to use for authentication to the remote_write API.
+  - `PASSWORD`: The password to use for authentication to the remote_write API.
```

Comment:
nit: for the example I would suggest to also set refresh_interval and endpoint + I would directly set some real looking data instead of the placeholders (the placeholders are already used in the ##usage part)

Reply:
Yes, I agree. I also don't like how here we repeat the definitions of the arguments. I did it this way because it's consistent with other discovery components, but I agree we should change this for all discovery components at a later point.